pipeline/splicing/calcVarPriors.py

```diff
@@ -595,27 +598,43 @@ def getVarLocation(variant, boundaries):
             return "UTR_variant"
         return "intron_variant"
 
-def getFastaSeq(chrom, rangeStart, rangeStop):
+def getFastaSeq(chrom, rangeStart, rangeStop, plusSeq=True):
     '''
     Given chromosome (in format 'chr13'), region genomic start position, and
     region genomic end position:
     Returns a string containing the sequence inclusive of rangeStart and rangeStop
-    To get + strand sequence, rangeStart < rangeStop
+    If plusSeq=True, returns plus strand sequence
+    If plusSeq=False, returns minus strand sequence
     '''
-    url = "http://togows.org/api/ucsc/hg38/%s:%d-%d.fasta" % (chrom, rangeStart, rangeStop)
+    if rangeStart < rangeStop:
+        regionStart = rangeStart
+        regionEnd = rangeStop
+    else:
+        regionStart = rangeStop
+        regionEnd = rangeStart
+    
+    url = "http://genome.ucsc.edu/cgi-bin/das/hg38/dna?segment=%s:%d,%d" % (chrom, regionStart, regionEnd)
     req = requests.get(url)
+    
+    if req.status_code == 429 and 'Retry-After' in req.headers:
+        retry = float(req.headers['Retry-After'])
+        time.sleep(retry)
+        req = requests.get(url)
+    
     lines = req.content.split('\n')
-
-    sequence = ""
-    for base in range(1, len(lines)):
-        sequence += lines[base]
-    return sequence
+    # because sequence is located at index 5 in dictionary
+    sequence = lines[5]
```

Comment:
Unfortunately there is not a header that gets returned. The sequence is the only thing in the list that isn't in brackets <>, so I could search for it in the list. Would that be better?

Reply:
in this case, it may be enough to put some sort of simple assertion around the sequence to ensure it's what you want and not something else. in this case, the script will fail if UCSC changes the index and the developer would find that in the assertion error. could be as simple as asserting the sequence is a string composed of only ['a', 'c', 't', 'g']